jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -262,6 +262,15 @@ void add_method_call_if_present(
   }
 }
 
+void add_static_method_call_if_present(
+  code_blockt &assignments,
+  const irep_idt &method_name,
+  const symbol_table_baset &symbol_table)
+{
+  add_method_call_if_present(
+    assignments, nil_exprt{}, method_name, symbol_table);
```

Comment:
:pick: I think it would be slightly nicer to use an optional `exprt` for the second argument, so we don't have to pass a `nil_exprt` like this. There could be a precondition that if the second argument is non-empty then the type has a `this` parameter.

Reply:
Passing in `null` as an instance to call a method on if the method is static is quite common in Java (e.g. `java.lang.reflect`) but I agree that an optionalt is probably clearer in the context of JBMC. So I changed it to that. I added an invariant that's the converse of your suggestion: if the type has a `this` parameter (i.e. it represents an instance method), the second argument has to be non-empty. It still seems safe to just ignore the second argument if the represented Java method is static. But if you think it's clearer I could change it to something like
```
    if(instance_expr)
    {
      INVARIANT(
        type.has_this(),
        "A call to a code_function_callt representing a static Java method "
        "should not be given an instance argument.");
      fun_call.arguments().push_back(address_of_exprt(*instance_expr));
    }
    else
    {
      INVARIANT(
        !type.has_this(),
        "A call to a code_function_callt representing a Java instance method "
        "should be given an instance argument.");
    }
```
Let me know what you think.

Reply:
Or I could just get rid of `instance_expr` altogether and make it part of the vector that @smowton suggested below? I'll try that.